### DISCUSSION TITLE
Get a folder causes 403 UnauthorizedAccessException

### DISCUSSION THREAD

**Discussion by krymancer** [pnp/pnpcore#1465] [Q&A] [answered]
Hello, I'm using PnP core to try and upload a file to a sharepoint site.

```C#
var folder = await context.Web.Folders.Where(f => f.Name == "Folder Name").FirstOrDefaultAsync();
```

I'm also using this scope in the controller

```C#
[AuthorizeForScopes(Scopes = new[] { "https://graph.microsoft.com/.default" })]
```

But this always return 403, I can get the site Title but not list the folders. Please help me

**Comment by jansenbe** [comment, marked as answer]
@Krymancer : this feature uses SharePoint REST under the covers, you need to grant the proper SharePoint scopes for your Entra app
